src/v/kafka/server/usage_aggregator.cc

```diff
@@ -181,6 +181,21 @@ usage_aggregator<clock_type>::usage_aggregator(
     _buckets[_current_window].reset(ss::lowres_system_clock::now());
 }
 
+template<typename clock_type>
+void usage_aggregator<clock_type>::rearm_window_timer() {
+    /// Calculate the next time the timer should fire, to adjust for skew ensure
+    /// the timer always fires at the top of the interval, weather it be hour,
+    /// minute, second, etc.
+    const auto now = clock_type::now();
+    const auto delta = std::chrono::seconds(
+      epoch_time_secs(now) % _usage_window_width_interval.count());
```

Comment:
nit: static assert that these are comparable (e.g. seconds % seconds)

Reply:
Please use `std::duration_cast`

Reply:
@BenPope even if the input is just plain int? From my understanding duration_cast is for casting one duration type to another

Reply:
The assert is fine, but I was thinking:

```c++
std::chrono::duration_cast<std::chrono::seconds>(_usage_window_width_interval).count()
```

Reply:
Even better, durations support `operator%`, so you don't have to keep erasing the type.

Reply:
I don't understand the comment about epoch time, but that should compile.
```c++
    const auto now = clock_type::now().time_since_epoch();
    /// This modulo trick only works because epoch time is hour aligned
    const auto delta = now % _usage_window_width_interval;
```